junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertionsTests.kt

```diff
@@ -62,12 +62,21 @@ class KotlinAssertionsTests {
     }
 
     @Test
-    fun `expected context exception testing`() = runBlocking<Unit> {
+    fun `assertThrows with coroutines`() = runBlocking<Unit> {
         assertThrows<AssertionError>("Should fail async") {
             suspend { fail("Should fail async") }()
         }
     }
 
+    @Test
+    fun `assertDoesNotThrow with coroutines`() = runBlocking {
+        val result = assertDoesNotThrow {
+            suspend { "Result" }()
```

Comment:
We might want to have another test with a suspending function that actually throws an exception (thus making the test fail), since we want to make sure that the assertion is actually waiting for the "throwing"

Reply:
Just noticed the test factory below, you could even add one dynamic test in each `dynamicContainer` for this new suspending function use case :)

Reply:
Done! https://github.com/junit-team/junit5/pull/2684/commits/d48eeb4afb14064648247f9f271ed4864b10285c